metadata-ingestion/src/datahub/ingestion/source/tableau.py

```diff
@@ -1371,27 +1371,36 @@ def _create_lineage_from_unsupported_csql(
             and tableau_constant.CONNECTION_TYPE in database
         ):
             upstream_tables = []
-            parser = LineageRunner(csql.get(tableau_constant.QUERY))
-
-            for table in parser.source_tables:
-                split_table = str(table).split(".")
-                if len(split_table) == 2:
-                    datset = make_table_urn(
-                        env=self.config.env,
-                        upstream_db=database.get(tableau_constant.NAME),
-                        connection_type=database.get(
-                            tableau_constant.CONNECTION_TYPE, ""
-                        ),
-                        schema=split_table[0],
-                        full_name=split_table[1],
-                        platform_instance_map=self.config.platform_instance_map,
-                        lineage_overrides=self.config.lineage_overrides,
-                    )
-                    upstream_tables.append(
-                        UpstreamClass(
-                            type=DatasetLineageType.TRANSFORMED, dataset=datset
+            query = csql.get(tableau_constant.QUERY)
+            parser = LineageRunner(query)
+
+            try:
+                for table in parser.source_tables:
+                    split_table = str(table).split(".")
+                    if len(split_table) == 2:
+                        datset = make_table_urn(
+                            env=self.config.env,
+                            upstream_db=database.get(tableau_constant.NAME),
+                            connection_type=database.get(
+                                tableau_constant.CONNECTION_TYPE, ""
+                            ),
+                            schema=split_table[0],
+                            full_name=split_table[1],
+                            platform_instance_map=self.config.platform_instance_map,
+                            lineage_overrides=self.config.lineage_overrides,
                         )
-                    )
+                        upstream_tables.append(
+                            UpstreamClass(
+                                type=DatasetLineageType.TRANSFORMED, dataset=datset
+                            )
+                        )
+            except Exception as e:
+                self.report.report_failure(
```

Comment:
maybe this should be a report_warning?

Reply:
Done in [ef727a3](https://github.com/datahub-project/datahub/pull/7965/commits/ef727a3b1df15d6f26467f904269d2da3a0d33e7)